code/mob/new_player.dm

```diff
@@ -397,16 +397,46 @@ mob/new_player
 			// probalby could be a define but dont give a shite
 			var/maxslots = 5
 			var/list/slots = list()
-			var/shown = clamp(c, (limit == -1 ? 99 : limit), maxslots)
+			var/shown = clamp(c, (limit == -1 ? maxslots : limit), maxslots)
 			// if there's still an open space, show a final join link
 			if (limit == -1 || (limit > maxslots && c < limit))
-				slots += "<a href='byond://?src=\ref[src];SelectedJob=\ref[J]' class='latejoin-card' style='border-color: [J.linkcolor];' title='Join the round as [J.name].'>&#x2713;&#xFE0E;</a>"
-
+				slots += {"<a href='byond://?src=\ref[src];
+				SelectedJob=\ref[J]' class='latejoin-card' style='border-color: [J.linkcolor];
+				' title='Join the round as [J.name].'>&#x2713;
+				&#xFE0E;
+				</a>"}
 			// show slots up to the limit
-			// extra people beyond the limit will be shown as a [+X] card
+			// extra people beyond the limit will be shown as a [+X] card, supposedly
 			for (var/i = shown, i > 0, i--)
-				slots += (i <= c ? "<div class='latejoin-card latejoin-full' style='border-color: [J.linkcolor]; background-color: [J.linkcolor];' title='Slot filled.'>[(i == 1 && c > shown) ? "+[c - maxslots]" : "&times;"]</div>" : "<a href='byond://?src=\ref[src];SelectedJob=\ref[J]' class='latejoin-card' style='border-color: [J.linkcolor];' title='Join the round as [J.name].'>&#x2713;&#xFE0E;</a>")
-
+				// can you believe all these slot appendages were in one line before using nested ternaries? awful.
+				if (i <= c)
+					if (i == 1 && c > shown)
+						slots += {"
+						<div
+						class='latejoin-card latejoin-full'
+						style='border-color: [J.linkcolor]; background-color: [J.linkcolor];'
+						title='Slot filled.'
+						>+[c - maxslots]
```

Comment:
and what about when it's infinite? this will display wrong

Reply:
Well when it's infinite, `limit` equals -1, but `maxslots` is made equal to `shown`, and is still capped at `5` (+ the free slot). Neither `maxslots` nor `c` should ever become infinite, so it works fine. From what i've tested using this code, it doesn't break either.
![unknown](https://user-images.githubusercontent.com/51260013/205489335-18140d98-b552-49e9-8106-f924abbd5d20.png)

The prudent change is in line 400:
`var/shown = clamp(c, (limit == -1 ? 99 : limit), maxslots)` to
`var/shown = clamp(c, (limit == -1 ? maxslots : limit), maxslots)`
the rest of the code changes are just cleaning it up to not extend off the right side. It uses exactly identical logic that the old system had before, just not in 200 character ternaries, but broken up into if else statements instead. If there is some logical error in there, it wasn't from me.